### PR TITLE
Wefork sandstorm update

### PR DESCRIPTION
**Pull request by dwrensha**
This is a rebase of https://github.com/wekan/wekan/pull/704, which gets the Sandstorm package working again and adds integration with Sandstorm notifications and in-grain sharing.

### CODE REVIEW

File: sandstorm-pkgdef.capnp [hunk 1]
```diff
@@ -184,6 +224,7 @@ const myCommand :Spk.Manifest.Command = (
   environ = [
     # Note that this defines the *entire* environment seen by your app.
     (key = "PATH", value = "/usr/local/bin:/usr/bin:/bin"),
+    (key = "SANDSTORM", value = "1"),
```
Comment (pierreozoux) [inline sandstorm-pkgdef.capnp:227]: I'm not familiar with Sandstorm, but is it really necessary for everybody?
(or if I say differently, I'm not a Sandstorm user, will this impact me?)

**Comment by pierreozoux** [comment]
I'd be glad if a sandstorm user could test it.
I quickly checked the code, and it looks good to me.

**Comment by ocdtrekkie** [comment]
As a useful note, there is a test link for this PR: https://oigimtlx81ukwop56zlo.oasis.sandstorm.io/

Discussion: https://groups.google.com/forum/#!topic/sandstorm-dev/26WWtYu0-kk

File: sandstorm-pkgdef.capnp (hunk 1 above)
Comment (ocdtrekkie) [reply sandstorm-pkgdef.capnp:227]: If you are not compiling for Sandstorm, the sandstorm-pkgdef.capnp file is never used. But, if you are compiling a Sandstorm SPK file, this setting in the package definition file will set up the environment variable SANDSTORM=1, which can be used elsewhere in the app to determine (if present) that the app should do something different for Sandstorm users.

So, for everyone else, this variable never appears, and so Sandstorm-specific code that checks for it won't run.

**Comment by pierreozoux** [comment]
Ok, I see this conversation related to (sorry, I'm also catching up with the project):
https://github.com/wekan/wekan/issues/415
https://github.com/wekan/wekan/issues/543

@AlexanderS could you comment on this PR please also?

I'm actually a bit concerned of seeing `sandstorm.js` growing. Just a quick question, for sandstorm notification, do you reuse notification facility that was created already in this PR:
https://github.com/wekan/wekan/pull/454

(Just to make sure code is not being duplicated, and that sandstorm functionalities match "traditional" functionality)

**Comment by dwrensha** [comment]
> do you reuse notification facility that was created already in this PR...

No, we do not use Wekan's built-in notification facility. Instead, we integrate on the level of _activity events_. We push all of Wekan's activity events to Sandstorm and leave it to Sandstorm to generate notifications from them. There's a bit of documentation about how this works in [activity.capnp](https://github.com/sandstorm-io/sandstorm/blob/v0.193/src/sandstorm/activity.capnp). The idea is that Sandstorm can provide a uniform, configurable interface for notifications across all grains -- even for grains of apps that don't independently implement notifications. Moreover, with the activity stream, Sandstorm can provide a uniform way to keep audit logs.

Note that the Wekan Sandstorm app already hides all of the UI elements that non-Sandstorm Wekan uses for configuring notifications.

**Comment by dwrensha** [comment]
> I'm actually a bit concerned of seeing sandstorm.js growing

The logic in sandstorm.js is only enabled when Wekan is running under Sandstorm. As with `sandstorm-pkgdef.capnp`, if you are not a Sandstorm user then it will not affect you.

We at Sandstorm have a vested interest in Wekan continuing to work well on our platform, and would be happy to take responsibility for maintaining these Sandstorm-specific files.

**Pull request merged**